File: com.neoba.dsync.vcdiff/src/com/neoba/dsync/vcdiff/Block.java
package com.neoba.dsync.vcdiff;

public class Block {

    String text;
    int offset;
    Block nextBlock;

    Block(String text, int offset) {
        this.text = text;
        this.offset = offset;
        this.nextBlock = null;
    }

    String getText() {
        return this.text;
    }

    int getOffset() {
        return this.offset;
    }

    void setNextBlock(Block nextBlock) {
        this.nextBlock = nextBlock;
    }
    
    Block getNextBlock() {
        return this.nextBlock;
    }
}

File: objective-c/vcdiff/Vcd/block.m
//
//  block.m
//  Vcd
//
//  Created by Akash K Sunny on 08/03/14.
//  Copyright 2014 Neoba Systems. All rights reserved.
//

#import "block.h"


@implementation block
-(void) initwith:(NSMutableString *) str:(int) off
{
	text=str;
	offset=off;
	nextblock=NULL;
	
}
-(NSMutableString *) gettext
{
	return text;
}

-(int) getoffset
{
	return offset;
}
-(void) setnextblock:(block *) b
{
	nextblock=b;
}
-(block *) getnextblock
{
	return nextblock;
}
-(id) copyWithZone: (NSZone *) zone 
{
block *newblock = [[block allocWithZone: zone] init];
[newblock initwith:text:offset]; 
	return newblock;
}
@end
